distsql/distsql_test.go

```diff
@@ -20,142 +20,93 @@ import (
 	"github.com/pingcap/parser/charset"
 	"github.com/pingcap/parser/mysql"
 	"github.com/pingcap/tidb/kv"
 	"github.com/pingcap/tidb/sessionctx"
+	"github.com/pingcap/tidb/sessionctx/stmtctx"
 	"github.com/pingcap/tidb/sessionctx/variable"
 	"github.com/pingcap/tidb/statistics"
 	"github.com/pingcap/tidb/types"
 	"github.com/pingcap/tidb/util/chunk"
 	"github.com/pingcap/tidb/util/codec"
+	"github.com/pingcap/tidb/util/disk"
 	"github.com/pingcap/tidb/util/execdetails"
 	"github.com/pingcap/tidb/util/memory"
+	"github.com/pingcap/tidb/util/mock"
 	"github.com/pingcap/tipb/go-tipb"
+	"github.com/stretchr/testify/require"
 	tikvstore "github.com/tikv/client-go/v2/kv"
 	"github.com/tikv/client-go/v2/tikv"
 	"github.com/tikv/client-go/v2/tikvrpc"
 )
 
-func (s *testSuite) createSelectNormal(batch, totalRows int, c *C, planIDs []int) (*selectResult, []*types.FieldType) {
-	request, err := (&RequestBuilder{}).SetKeyRanges(nil).
-		SetDAGRequest(&tipb.DAGRequest{}).
-		SetDesc(false).
-		SetKeepOrder(false).
-		SetFromSessionVars(variable.NewSessionVars()).
-		SetMemTracker(memory.NewTracker(-1, -1)).
-		Build()
-	c.Assert(err, IsNil)
-
-	// 4 int64 types.
-	colTypes := []*types.FieldType{
-		{
-			Tp:      mysql.TypeLonglong,
-			Flen:    mysql.MaxIntWidth,
-			Decimal: 0,
-			Flag:    mysql.BinaryFlag,
-			Charset: charset.CharsetBin,
-			Collate: charset.CollationBin,
-		},
-	}
-	colTypes = append(colTypes, colTypes[0])
-	colTypes = append(colTypes, colTypes[0])
-	colTypes = append(colTypes, colTypes[0])
-
-	// Test Next.
-	var response SelectResult
-	if planIDs == nil {
-		response, err = Select(context.TODO(), s.sctx, request, colTypes, statistics.NewQueryFeedback(0, nil, 0, false))
-	} else {
-		response, err = SelectWithRuntimeStats(context.TODO(), s.sctx, request, colTypes, statistics.NewQueryFeedback(0, nil, 0, false), planIDs, 1)
-	}
-
-	c.Assert(err, IsNil)
-	result, ok := response.(*selectResult)
-	c.Assert(ok, IsTrue)
-	c.Assert(result.label, Equals, "dag")
-	c.Assert(result.sqlType, Equals, "general")
-	c.Assert(result.rowLen, Equals, len(colTypes))
-
-	resp, ok := result.resp.(*mockResponse)
-	c.Assert(ok, IsTrue)
-	resp.total = totalRows
-	resp.batch = batch
-
-	return result, colTypes
-}
-
-func (s *testSuite) TestSelectNormal(c *C) {
-	response, colTypes := s.createSelectNormal(1, 2, c, nil)
+func TestSelectNormal(t *testing.T) {
+	t.Parallel()
+	response, colTypes := createSelectNormal(t, 1, 2, nil, nil)
 
 	// Test Next.
 	chk := chunk.New(colTypes, 32, 32)
 	numAllRows := 0
 	for {
 		err := response.Next(context.TODO(), chk)
-		c.Assert(err, IsNil)
+		require.NoError(t, err)
 		numAllRows += chk.NumRows()
 		if chk.NumRows() == 0 {
 			break
 		}
 	}
-	c.Assert(numAllRows, Equals, 2)
-	err := response.Close()
-	c.Assert(err, IsNil)
-	c.Assert(response.memTracker.BytesConsumed(), Equals, int64(0))
+	require.Equal(t, 2, numAllRows)
+	require.NoError(t, response.Close())
+	require.EqualValues(t, 0, response.memTracker.BytesConsumed())
```

Comment:
I prefer `Equal(t, int64(0), response.memTracker.BytesConsumed())`, the orginal check is more strict/of powerful strength. Type changes in the function signature will be exposed if any. So I suggest `Equal`, same to other checks.

Reply:
I understand, here we need to constrain the type in the test, let me fix it.